cpp/include/cudf/interop.hpp

```diff
@@ -150,10 +150,11 @@ struct column_metadata {
  * 9 which is the maximum precision for 32-bit types. Similarly, numeric::decimal128 will be
  * converted to Arrow decimal128 of the precision 38.
  */
-std::shared_ptr<arrow::Table> to_arrow(table_view input,
-                                       std::vector<column_metadata> const& metadata = {},
-                                       rmm::cuda_stream_view stream = cudf::get_default_stream(),
-                                       arrow::MemoryPool* ar_mr     = arrow::default_memory_pool());
+[[deprecated]] std::shared_ptr<arrow::Table> to_arrow(
```

Comment:
Could we also update the documentation to point to the appropriate new APIs for to/from? [Our developer guide has some notes on the way to do that with doxygen](https://docs.rapids.ai/api/libcudf/nightly/developer_guide).

Reply:
Done in https://github.com/rapidsai/cudf/pull/16297/commits/11e0601097163cbec51485d6d6584a3756b15daf.